include/drivers/usbc/usbc_tcpc.h

```diff
@@ -0,0 +1,1020 @@
+/* usbc_tcpc.h - USB Type-C Port Controller driver interface */
+
+/*
+ * Copyright 2022 The Chromium OS Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief USBC Type-C Port Controller device APIs
+ *
+ * This file contains the USB Type-C Port Controller device APIs.
+ * All Type-C Port Controller device drivers should implement the
+ * APIs described in this file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_USBC_USBC_TCPC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_USBC_USBC_TCPC_H_
+
+/**
+ * @brief USB Type-C Port Controller API
+ * @defgroup _usb_type_c_port_controller_api USB Type-C Port Controller API
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#include "usbc_tc.h"
+#include "usbc_pd.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief TCPC Alert bits
+ */
+enum tcpc_alert {
+	/* CC status changed */
+	TCPC_ALERT_CC_STATUS,
+	/* Power status changed */
+	TCPC_ALERT_POWER_STATUS,
+	/* Receive Buffer register changed */
+	TCPC_ALERT_MSG_STATUS,
+	/* Received Hard Reset message */
+	TCPC_ALERT_HARD_RESET_RECEIVED,
+	/* SOP* message transmission not successful */
+	TCPC_ALERT_TRANSMIT_MSG_FAILED,
+	/*
+	 * Reset or SOP* message transmission not sent
+	 * due to an incoming receive message
+	 */
+	TCPC_ALERT_TRANSMIT_MSG_DISCARDED,
+	/* Reset or SOP* message transmission successful */
+	TCPC_ALERT_TRANSMIT_MSG_SUCCESS,
+	/* A high-voltage alarm has occurred */
+	TCPC_ALERT_VBUS_ALARM_HI,
+	/* A low-voltage alarm has occurred */
+	TCPC_ALERT_VBUS_ALARM_LO,
+	/* A fault has occurred. Read the FAULT_STATUS register */
+	TCPC_ALERT_FAULT_STATUS,
+	/* TCPC RX buffer has overflowed */
+	TCPC_ALERT_RX_BUFFER_OVERFLOW,
+	/* The TCPC in Attached.SNK state has detected a sink disconnect */
+	TCPC_ALERT_VBUS_SNK_DISCONNECT,
+	/* Receive buffer register changed */
+	TCPC_ALERT_BEGINNING_MSG_STATUS,
+	/* Extended status changed */
+	TCPC_ALERT_EXTENDED_STATUS,
+	/*
+	 * An extended interrupt event has occurred. Read the alert_extended
+	 * register
+	 */
+	TCPC_ALERT_EXTENDED,
+	/* A vendor defined alert has been detected */
+	TCPC_ALERT_VENDOR_DEFINED
+};
+
+/**
+ * @brief TCPC Status register
+ */
+enum tcpc_status_reg {
+	/* The CC Status register */
+	TCPC_CC_STATUS,
+	/* The Power Status register */
+	TCPC_POWER_STATUS,
+	/* The Fault Status register */
+	TCPC_FAULT_STATUS,
+	/* The Extended Status register */
+	TCPC_EXTENDED_STATUS,
+	/* The Extended Alert Status register */
+	TCPC_EXTENDED_ALERT_STATUS,
+	/* The Vendor Defined Status register */
+	TCPC_VENDOR_DEFINED_STATUS
+};
+
+/**
+ * @brief TCPC Chip Information
+ */
+struct tcpc_chip_info {
+	/* Vendor Id */
+	uint16_t vendor_id;
+	/* Product Id */
```

Comment:
Please use Doxygen docstrings everywhere:

```suggestion
	/** Vendor Id */
	uint16_t vendor_id;
	/** Product Id */
```

Reply:
Done